chainable/__init__.py

```diff
@@ -0,0 +1,46 @@
+from functools import reduce, _initial_missing
+from typing import TypeVar, Callable, overload, Generic
+from collections.abc import Iterable, Iterator
+
+__all__ = ["Chainable"]
+
+T = TypeVar("T")
+S = TypeVar("S")
+
+def flat_map_helper(self: "Chainable[T]", mapper: Callable[[T], Iterable[S]]) -> Iterator[S]:
+    for val in self:
+        yield from mapper(val)
+    
+
+class Chainable(Generic[T]):
```

Comment:
I'm not loving "chainable". It's certainly acceptable and more than adequate, but I wonder if there's an even more concise term or moniker that could suit. If there was a name that was just one or two syllables and fewer characters, it would be less intrusive to wrap. Some (stream of consciousness) ideas:

- `Lump` - lumps together the operations
- `Ring` - a precious loop of things
- `Gem` - also precious
- `Direct` - a linear interface to otherwise spiraling logic
- `Taut` - spiraling logic pulled taught (straight)

I'm liking Taut enough that I've stalled on more ideas.

Reply:
I generally try to pick names that are no more than acceptable in hopes that someone else will come up with a better name. I think I like `Taut` as well, though not sure about the library name.

Reply:
After having a rest, I'm less reluctant about Chainable. Having a consistent package name is nice; it'd be a shame to lose that. And if a user wants a short name, they can create one. Feel free to dismiss this concern.

It does look like [taut](https://pypi.org/project/taut/) is available; just currently blocked for "security purposes". Since this would be a legitimate purpose, I'd expect them to transfer the project. If you like the name enough, I'd be happy to help negotiate getting that name transferred.